### PR TITLE
CORE-1016: initial event recorder implementation

### PR DESCRIPTION
**Pull request by slr71**
I believe that this is done, but I want to proofread the code and leave some comments in the PR before anyone offers suggestions.

### CODE REVIEW

File: common/times.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	// If the original is empty, simply return it.
+	if original == "" {
+		return original, nil
+	}
+
+	// If the original already appears to be an epoch time, simply return it.
+	epochRegexp := regexp.MustCompile("^\\d+$")
+	if epochRegexp.MatchString(original) {
+		return original, nil
+	}
```
Comment (slr71) [inline common/times.go:31]: I technically could have done a more detailed check here to see if the date seemed reasonable, but these timestamps are coming from other DE microservices. It didn't seem worthwhile to expend the effort needed to do anything more detailed.

File: common/times.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	// Attempt to parse the timestamp. We only support one format for now.
+	timestamp, err := time.Parse(time.RFC3339Nano, original)
+	if err != nil {
+		return "", errors.Wrap(err, wrapMsg)
+	}
```
Comment (slr71) [inline common/times.go:37]: RFC3339Nano will also parse plain RFC3339 timestamps, so this should work well for us.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	github.com/cyverse-de/logcabin v0.0.0-20200527185931-2ccd850e39ad
+	github.com/cyverse-de/messaging v6.0.0+incompatible
+	github.com/lib/pq v1.7.0
+	github.com/mcnijman/go-emailaddress v1.1.0
+	github.com/pkg/errors v0.8.1
+	github.com/spf13/viper v1.7.0 // indirect
+	github.com/streadway/amqp v1.0.0
+)
+
+replace github.com/cyverse-de/messaging => /Users/sarahr/src/de/go/src/github.com/cyverse-de/messaging
```
Comment (slr71) [inline go.mod:19]: I'll remove this after the updates to the `messaging` library have been merged.

**Review by slr71 — COMMENTED** [review]
I deviated a little bit from the design document. I'll make some updates to the design document today.

File: common/times.go (hunk 1 above)
Comment (ianmcorvidae) [reply common/times.go:31]: I suppose anything that's all numbers is a valid epoch time regardless, so I think it's fine, worst that happens is we get some very silly notifications from 1970 probably

**Review by ash3rz — APPROVED** [review]
Not at all my area of expertise, but I didn't catch anything 👍 
Code seemed really clean and easy to follow

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**